src/main/java/uk/nhs/tis/trainee/actions/service/ActionService.java

```diff
@@ -172,6 +175,24 @@ public List<ActionDto> updateActions(Operation operation, ProgrammeMembershipDto
       deleteIncompleteActions(action);
     }
 
+    // Handle Conditions of Joining (CoJ) action. We ignore ACTIONS_EPOCH and start date here to
+    // avoid dangling CoJ actions where a programme start date has been updated to before the epoch.
+    if (Objects.equals(operation, Operation.LOAD) && dto.conditionsOfJoining() != null) {
+      log.info("Completing any CoJ actions for Programme Membership {}.", dto.id());
+      // If a CoJ action has just been created, replace it with a new completed action. This should
+      // only be possible if bulk-loading programme memberships that already have signed CoJs.
+      Optional<Action> addedAction = actions.stream()
+          .filter(a -> a.type().equals(SIGN_COJ)).findFirst();
+      if (addedAction.isPresent()) {
+        actions.remove(addedAction.get());
+        actions.add(mapper.complete(addedAction.get()));
```

Comment:
'complete' will set the `completed` instant to now(), as it does for any other action. It might make more sense to rather set `completed = syncedAt`. Generally (in the FE at least) `completed` is treated more like a boolean than read for its specific value, so it doesn't matter so much. But in future, or for reporting, we might care more about when it was completed. Thoughts?

Reply:
Edit: talked myself around to set `completed = syncedAt`